persistent/tests/test_persistence.py

```diff
@@ -166,6 +228,14 @@ def _test():
             inst._p_oid = new_OID
         self.assertRaises(ValueError, _test)
 
+    def test_assign_p_oid_not_in_cache_allowed(self):
+        jar = self._makeJar()
+        inst = self._makeOne()
+        inst._p_jar = jar
+        inst._p_oid = 1 # anything goes
+        inst._p_oid = 42
+        self.assertEqual(inst._p_oid, 42)
```

Comment:
>> (True, the picklecache has that requirement, but it's arguable that the picklecache is ZODB or storage specific.)

> How can you make practical use of Persistent without also using picklecache? The whole point of the class is to have changes to an object get registered with the cache.

I suspect you meant to write "get registered with the connection", aka `_p_jar`. And any caching that's done is the purview of the connection. Alternate `_p_jar` implementations may use a different cache implementation, or none at all.